training-app/app/models/book.rb

```diff
@@ -0,0 +1,4 @@
+# frozen_string_literal: true
+
+class Book < ApplicationRecord
```

Comment:
Add validations for not null in the model as well. If you only have them on the DB when you get a DB constraint exception it won't have as much information as if a rails validation fails.